posthog/hogql/database/database.py

```diff
@@ -572,27 +570,34 @@ def serialize_database(
         fields_dict = {field.name: field for field in fields}
         tables[table_key] = DatabaseSchemaPostHogTable(fields=fields_dict, id=table_key, name=table_key)
 
-    # Data Warehouse Tables
+    # Data Warehouse Tables and Views - Fetch all related data in one go
     warehouse_table_names = context.database.get_warehouse_tables()
-    warehouse_tables = (
-        list(
-            DataWarehouseTable.objects.select_related("credential", "external_data_source")
-            .filter(Q(deleted=False) | Q(deleted__isnull=True), team_id=context.team_id, name__in=warehouse_table_names)
-            .all()
-        )
-        if len(warehouse_table_names) > 0
-        else []
-    )
-    warehouse_schemas = (
-        list(
-            ExternalDataSchema.objects.exclude(deleted=True)
-            .filter(team_id=context.team_id, table_id__in=[table.id for table in warehouse_tables])
-            .all()
+    views = context.database.get_views()
+
+    # Fetch warehouse tables with related data in a single query
+    warehouse_tables_with_data = (
+        DataWarehouseTable.objects.select_related("credential", "external_data_source")
```

Comment:
i'm pretty rust on exactly what django ORM spits out but this should be ultimately better than n+3 queries